### PR TITLE
Clean up possibly unnecessary code

### PR DESCRIPTION
**Pull request by sudhirj**

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -102,35 +101,12 @@ func (rhc *RangingHTTPClient) StandardClient() *http.Client {
...
-	// The client to use during requests. If nil, a default ranging client is used.
 	RangingClient *RangingHTTPClient
-
-	// once ensures that the logic to initialize the default client runs at
-	// most once, in a single thread.
-	once sync.Once
-}
-
-// init initializes the underlying ranging client.
-func (rt *RoundTripper) init() {
```
Comment (sudhirj) [inline http.go:114]: @ananthb Why is this necessary? Zero values are pointless in this case. Client absolutely needs configuration, the chunk size and parallelism are necessary to do anything.

File: http.go [hunk 2]
```diff
@@ -102,35 +101,12 @@ func (rhc *RangingHTTPClient) StandardClient() *http.Client {
...
 
 // RoundTrip satisfies the http.RoundTripper interface.
 func (rt *RoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
-	rt.once.Do(rt.init)
-
-	// Execute the request.
-	resp, err := rt.RangingClient.Do(req)
-	// If we got an error returned by standard library's `Do` method, unwrap it
-	// otherwise we will wind up erroneously re-nesting the error.
-	if _, ok := err.(*url.Error); ok {
```
Comment (sudhirj) [inline http.go:129]: Explain this a bit more? Where would we re-wrap this? Maybe add a test as well to establish what's going on?

Comment (ananthb) [reply http.go:129]: It's from a similar HTTP transport implementation. ranger only needs it in case it wraps errors, but I don't think that applies here.
